benchmark.py

```diff
@@ -268,18 +297,36 @@ def _run_benchmark(bazel_binary_path,
         '--build_event_json_file=%s' % bep_json_path
     ]
 
-    _single_run(bazel_binary_path, command, command_args, bazelrc,
-                collect_memory)
+    _single_run(
+        bazel_binary_path, command, command_args, bazelrc, collect_memory)
     command, expressions, options = args_parser.parse_bazel_args_from_build_event(
         bep_json_path)
   else:
     logger.log('Parsing arguments from command line...')
     command, expressions, options = args_parser.parse_bazel_args_from_canonical_str(
         bazel_args)
 
-  parsed_args = options + expressions
+  if collect_json_profile:
+    assert data_directory
+    if not os.path.exists(data_directory):
+      os.makedirs(data_directory)
+
   for i in range(runs):
     logger.log('Starting benchmark run %s/%s:' % ((i + 1), runs))
+
+    maybe_include_json_profile_flags = options[:]
+    if collect_json_profile:
+      assert bazel_commit
+      assert project_commit
+      maybe_include_json_profile_flags += _construct_json_profile_flags(
+          '%s/%s_%s_%s_%s_of_%s.profile.gz' % (
+              data_directory,
+              bazel_bench_uid,
+              bazel_commit,
+              project_commit,
+              i + 1,
```

Comment:
nit: instead of the + 1 here, you could also change the range/call-site

Reply:
Done